pkg/taxonomy/model/additionalprops.go

```diff
@@ -41,3 +45,22 @@ func (a AdditionalPropertiesType) MarshalJSON() ([]byte, error) {
 
 	return json.Marshal(a.Schema)
 }
+
+func (a *AdditionalPropertiesType) ToJSONSchemaProps(flattenBy *Document) *apiextensions.JSONSchemaPropsOrBool {
+	if a == nil {
+		return nil
+	}
+
+	if a.Schema != nil {
+		return &apiextensions.JSONSchemaPropsOrBool{Allows: true, Schema: a.Schema.ToJSONSchemaProps(flattenBy)}
+	}
+
+	// additionalProperties: false is forbidden in Kubernetes
+	// x-kubernetes-preserve-unknown-fields replaces the functionality of additionalProperties: true
+	if a.Allowed != nil {
+		// return &apiextensions.JSONSchemaPropsOrBool{Allows: *a.Allowed}
+		return nil
```

Comment:
Shouldn't it be `return &apiextensions.JSONSchemaPropsOrBool{Allows: true}` ?

Reply:
For validation we remove additionalProperties unless it's set to a concrete schema (because of the reasons in the comment above this if condition)